### PR TITLE
fix and extend e-mail matching in customer-admin-controller

### PR DESCRIPTION
**Pull request by jim-minter**
```release-note
Resolve customer-admin group memberships issue when case of AAD mail domain differs from case of AAD login
```

### CODE REVIEW

**Comment by jim-minter** [comment]
/hold
to see if this fixes customer issue

@mjudeikis ptal
@thekad @Makdaam fyi

**Comment by Makdaam** [comment]
I missed the AADUser.Mail return. :(
lgtm

File: pkg/controllers/customeradmin/group_mapping_test.go [hunk 1]
```diff
@@ -309,6 +309,38 @@ func TestFromMSGraphGroup(t *testing.T) {
...
+			},
+			want: &v1.Group{
+				ObjectMeta: meta_v1.ObjectMeta{
+					Name: osaCustomerAdmins,
+				},
+				Users: []string{"user@home.com"},
+			},
+		},
+		{
+			name:          "user case can't match",
```
Comment (ehashman) [inline pkg/controllers/customeradmin/group_mapping_test.go]: I don't really understand this test case, how is it different than e.g. the one on L234?

**Comment by mjudeikis** [comment]
from the context, I assume this is @Makdaam prod incident 
/lgtm

File: pkg/controllers/customeradmin/group_mapping_test.go (hunk 1 above)
Comment (jim-minter) [reply pkg/controllers/customeradmin/group_mapping_test.go]: previously in some cases we were incorrectly doing case insensitive match on the whole e-mail address.  In fact the domain part is case insensitive and the username part is case sensitive.  The first additional test case checks that we correctly coalesce Home.com -> home.com; the second checks that we do not incorrectly coalesce User@ to user@.

**Pull request merged**